java/test.sh

```diff
@@ -8,7 +8,10 @@ set -x
 ROOT_DIR=$(cd "$(dirname "${BASH_SOURCE:-$0}")"; pwd)
 
 run_testng() {
+    # The command `$@` will exited immediately once `set -e`.
+    set +e
     $@
+    set -e
     local exit_code=$?
```

Comment:
Ah okay sorry, I didn't realize this needs to keep going in that case. Can we do the following instead?

```suggestion
    local exit_code
    if $@; then
        exit_code=0
    else
        exit_code=$?
    fi
```

This should preserve the exit code without exiting the script.

Reply:
Great.